File: app/src/test/java/io/github/freewebmovement/igniter/StorageTest.java
package io.github.freewebmovement.igniter;

import static junit.framework.TestCase.assertEquals;

import android.content.Context;

import androidx.test.runner.AndroidJUnit4;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import io.github.freewebmovement.igniter.persistence.Storage;

@RunWith(AndroidJUnit4.class)
public class StorageTest {
     Context instrumentationContext;

    @Before
    public void setup() {
        instrumentationContext =  androidx.test.core.app.ApplicationProvider.getApplicationContext();
    }
    @Test
    public void shouldInit() {

        Storage storage = new Storage(instrumentationContext);

        String[] paths = {
                storage.getCaCertPath(),
                storage.getCountryMmdbPath(),
                storage.getClashConfigPath()
        };

        int[] ids = {
                R.raw.cacert,
                R.raw.country,
                R.raw.clash_config
        };

        storage.reset();

        for(int i = 0; i < paths.length; i++) {
            String filename = paths[i];
            String rawString = storage.readRawText(ids[i]);
            String content = new String(Storage.read(filename));
            assertEquals(content, rawString);
        }
    }
}

File: app/src/test/java/io/github/freewebmovement/igniter/StorageTest.kt
package io.github.freewebmovement.igniter

import android.content.Context
import androidx.test.core.app.ApplicationProvider
import androidx.test.runner.AndroidJUnit4
import io.github.freewebmovement.igniter.persistence.Path
import io.github.freewebmovement.igniter.persistence.Storage
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class StorageTest {
    private lateinit var instrumentationContext: Context

    @Before
    fun setup() {
        instrumentationContext = ApplicationProvider.getApplicationContext()
    }

    @Test
    fun shouldInit() {
        val storage = Storage(instrumentationContext)
        val path: Path = storage.path

        val paths = arrayOf(path.caCert, path.countryMmdb, path.clashConfig)

        val ids = intArrayOf(R.raw.cacert, R.raw.country, R.raw.clash_config)

        storage.reset()

        for (i in paths.indices) {
            val filename = paths[i]!!
            val rawString = storage.readRawText(ids[i])
            val content = String(Storage.read(filename)!!, Charsets.UTF_8)
            assertEquals(content, rawString)
        }
    }
}
